### DISCUSSION TITLE
How to create a reuseable Popup

### DISCUSSION THREAD

**Discussion by prabhavmehra** [CommunityToolkit/Maui#1703] [Q&A]
How to create a reusable styled popup? I want to define the border radius and some other style properties and want other popups to inherit from this `BasePopup`. 

We can do this using ControlTemplate and ContentPresenter in a ContentPage but I don't think it's possible here.

**Comment by bijington** [comment]
Does this section from our docs help?

https://learn.microsoft.com/en-gb/dotnet/communitytoolkit/maui/views/popup#styling

**Reply by prabhavmehra** [reply]
I did have a look at this but this doesn’t provide the required styling. For instance I am looking at a corner radius for rounded borders. 

The other thing I which wanted to have was two buttons with fixed styles applied and let the popup inheriting this provide commands as bindable property

**Reply by bijington** [reply]
To consistently apply a CornerRadius on a Popup I think you need to place a Border inside. Therefore you won't be styling the Popup but the Border instead.

The addition of the Buttons makes me ask why don't you just create a single Popup implementation and add your own ContentTemplate property that you can set for each usage?

**Reply by prabhavmehra** [reply]
I have applied a corner radius to the border but that’s more consistent with the app styling and is different to what the popup needs. 

Is there an example I can see for the second suggestion?

**Reply by bijington** [reply]
Off the top of my head no. Out of interest is the problem you are trying to solve, showing multiple different popups with different content but a similar style?

**Reply by bijington** [reply]
Also you can mark a Style with `x:Key` and then use it explicitly in your popup. This allows you to define one style for borders specifically for your popups

**Reply by prabhavmehra** [reply]
Yes that is what I want to do.

**Comment by CliffAgius** [comment]
Closed as answered, if closed in error please let us know to re-open.
